runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/WorkItemKeySelector.java

```diff
@@ -49,6 +52,6 @@ public ByteBuffer getKey(WindowedValue<SingletonKeyedWorkItem<K, V>> value) thro
 
   @Override
   public TypeInformation<ByteBuffer> getProducedType() {
-    return new GenericTypeInfo<>(ByteBuffer.class);
+    return new CoderTypeInformation<>(FlinkKeyUtils.ByteBufferCoder.of(), pipelineOptions.get());
```

Comment:
This looks unrelated, can you just explain this modification? I suppose it is correct, just wonder why it was `GenericTypeInfo` before.

Reply:
Not sure if this is necessary. I wanted to ensure that the new "reinterpreted partitioning" is compatible with the one used by GBK / Combine.

The idea was if partitioning is not compatible, it may result in some state partitioning related glitches (eg. you wouldn't have local state for a key-group you need).

Second thoughts, flink selects target partition based on "pojo hash code" (not based on binary representation), so the previous version was probably compatible enough 🤔 

@mxm WDYT?

Reply:
I would think that the partitioning is ensured by the `reinterpretAsKeyedStream` and will therefore be preserved from the previous shuffle phase. Is this not enough?

Reply:
I think that `reinterpretAsKeyedStream` just trusts the user and there are no safety nets in place.